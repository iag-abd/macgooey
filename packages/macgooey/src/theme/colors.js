import * as R from "ramda";

export const colorOrShade = (defaultShade = "lightGray") => p =>
  R.pathOr(
    R.pathOr("lightGray", ["theme", "colorShades", defaultShade], p),
    ["theme", "color", p.color],
    p
  );

export const complimentOrShade = (defaultShade = "lightGray") => p =>
  R.pathOr(
    R.pathOr("lightGray", ["theme", "colorShades", defaultShade], p),
    ["theme", "colorCompliment", p.color],
    p
  );
