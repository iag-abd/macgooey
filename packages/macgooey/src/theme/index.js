// @flow

export { ThemeProvider } from "styled-components";

export type ColorPalette = {|
  primary: string,
  secondary: string,
  tertiary: string
|};

export type ColorShade = {|
  lighterGray: string,
  lightGray: string,
  gray: string,
  darkGray: string,
  darkerGray: string
|};

type ElementStyles = {|
  borderRadius: string,
  borderWidth: string
|};

type ButtonStyles = ElementStyles;

type InputStyles = ElementStyles;

export type Styles = {|
  button: ButtonStyles,
  color: ColorPalette,
  colorCompliment: ColorPalette,
  colorShade: ColorShade,
  fontFamilyHeading: string,
  input: InputStyles,
  textColorBody: string
|};

export const themeBase: Styles = {
  fontFamilyHeading: "'Open Sans', sans-serif",
  textColorBody: "#333",

  button: {
    borderRadius: "3px",
    borderWidth: "0"
  },

  input: {
    borderWidth: "1px",
    borderRadius: "0"
  },

  color: {
    primary: "#4375E8",
    secondary: "#FF4448",
    tertiary: "#E8B221"
  },
  colorCompliment: {
    primary: "#FFF",
    secondary: "#FFF",
    tertiary: "#FFF"
  },
  colorShade: {
    lighterGray: "#eee",
    lightGray: "#ddd",
    gray: "#aaa",
    darkGray: "#555",
    darkerGray: "#222"
  }
};

export const themeAerith = {
  ...themeBase,

  fontFamilyHeading: "'Montserrat', sans-serif",
  textColorBody: "#414042",

  input: {
    borderWidth: "1px",
    borderRadius: "0"
  },

  color: {
    default: "#DDD",
    primary: "#0BAB47",
    secondary: "#6D6E71",
    tertiary: "#007BAD"
  },
  colorCompliment: {
    default: "#333",
    primary: "#FFF",
    secondary: "#FFF",
    tertiary: "#FFF"
  }
};

export const themeSephiroth = {
  ...themeBase,

  fontFamilyHeading: "'Source Sans Pro', 'Open Sans', sans-serif",
  textColorBody: "#3B3D40",

  button: {
    borderRadius: "3px",
    borderWidth: "1px"
  },

  color: {
    default: "#F0F0F0",
    primary: "#602f89",
    secondary: "#01B5CC",
    tertiary: "#1E2741"
  },
  colorCompliment: {
    default: "#602F89",
    primary: "#FFF",
    secondary: "#FFF",
    tertiary: "#FFF"
  }
};
