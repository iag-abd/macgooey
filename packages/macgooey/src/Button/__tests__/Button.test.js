import React from "react";
import { Button } from "..";
import "jest-styled-components";

import renderer from "react-test-renderer";
import { themeBase } from "../../theme";

describe("<Button />", () => {
  test("it should render", () => {
    const component = renderer.create(
      <Button theme={themeBase}>Test Button</Button>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  test("it should render the primary button", () => {
    const component = renderer.create(
      <Button theme={themeBase} color="primary">
        Test Button
      </Button>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  test("it should render the secondary button", () => {
    const component = renderer.create(
      <Button theme={themeBase} color="secondary">
        Test Button
      </Button>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  test("it should render the tertiary button", () => {
    const component = renderer.create(
      <Button theme={themeBase} color="tertiary">
        Test Button
      </Button>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
