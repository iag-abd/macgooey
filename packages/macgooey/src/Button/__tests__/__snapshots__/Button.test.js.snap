// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it should render 1`] = `
.c0 {
  border-radius: 3px;
  border: none;
  font-family: 'Open Sans',sans-serif;
  font-size: 0.9em;
  padding: 12px 14px;
  cursor: pointer;
  outline: none;
  -webkit-transition: background 150ms;
  transition: background 150ms;
  background: #eee;
  border-color: #d5d5d5;
  border-style: solid;
  border-width: 0;
  color: #555;
}

.c0:hover {
  background: #d5d5d5;
}

<button
  className="c0"
  color="default"
>
  Test Button
</button>
`;

exports[`<Button /> it should render the primary button 1`] = `
.c0 {
  border-radius: 3px;
  border: none;
  font-family: 'Open Sans',sans-serif;
  font-size: 0.9em;
  padding: 12px 14px;
  cursor: pointer;
  outline: none;
  -webkit-transition: background 150ms;
  transition: background 150ms;
  background: #4375E8;
  border-color: #1b56dd;
  border-style: solid;
  border-width: 0;
  color: #FFF;
}

.c0:hover {
  background: #1b56dd;
}

<button
  className="c0"
  color="primary"
>
  Test Button
</button>
`;

exports[`<Button /> it should render the secondary button 1`] = `
.c0 {
  border-radius: 3px;
  border: none;
  font-family: 'Open Sans',sans-serif;
  font-size: 0.9em;
  padding: 12px 14px;
  cursor: pointer;
  outline: none;
  -webkit-transition: background 150ms;
  transition: background 150ms;
  background: #FF4448;
  border-color: #ff1116;
  border-style: solid;
  border-width: 0;
  color: #FFF;
}

.c0:hover {
  background: #ff1116;
}

<button
  className="c0"
  color="secondary"
>
  Test Button
</button>
`;

exports[`<Button /> it should render the tertiary button 1`] = `
.c0 {
  border-radius: 3px;
  border: none;
  font-family: 'Open Sans',sans-serif;
  font-size: 0.9em;
  padding: 12px 14px;
  cursor: pointer;
  outline: none;
  -webkit-transition: background 150ms;
  transition: background 150ms;
  background: #E8B221;
  border-color: #c29314;
  border-style: solid;
  border-width: 0;
  color: #FFF;
}

.c0:hover {
  background: #c29314;
}

<button
  className="c0"
  color="tertiary"
>
  Test Button
</button>
`;
